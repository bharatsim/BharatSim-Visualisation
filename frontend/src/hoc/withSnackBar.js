import React from 'react';
import { SnackbarProvider } from 'notistack';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import failureIcon from '../assets/images/error-icon.svg';
import successIcon from '../assets/images/success-icon.svg';

const useSnackBarStyles = makeStyles((theme) => ({
  contentRoot: {
    padding: theme.spacing(1, 4),
  },
  message: {
    padding: 0,
  },
  variantSuccess: {
    backgroundColor: `#EAF4E5 !important`,
    borderColor: `${theme.palette.success.main}33`,
    border: '1px solid',
    color: `${theme.palette.success.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
  variantError: {
    backgroundColor: `#FBE9E4 !important`,
    borderColor: `${theme.palette.error.light}33`,
    border: '1px solid',
    color: `${theme.palette.error.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
}));

const useIconStyles = makeStyles((theme) => ({
  iconRoot: {
    marginRight: theme.spacing(2),
    height: theme.spacing(10),
    width: theme.spacing(10),
  },
}));

function withSnackBar(WrappedComponent) {
  return (props) => {
    const snackBarClasses = useSnackBarStyles();
    const iconClasses = useIconStyles();
    const snackBarRef = React.createRef();
    const onClickDismiss = (key) => () => {
      snackBarRef.current.closeSnackbar(key);
    };
    return (
      <SnackbarProvider
        maxSnack={3}
        ref={snackBarRef}
        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
        classes={snackBarClasses}
        iconVariant={{
          error: <img src={failureIcon} alt="error" className={iconClasses.iconRoot} />,
          success: <img src={successIcon} alt="success" className={iconClasses.iconRoot} />,
        }}
        action={(key) => <Button onClick={onClickDismiss(key)}>Dismiss</Button>}
      >
        <WrappedComponent {...props} />
      </SnackbarProvider>
    );
  };
}

export default withSnackBar;
