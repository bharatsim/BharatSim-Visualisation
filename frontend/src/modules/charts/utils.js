import { chartStyleConfig, getColor } from './chartStyleConfig';

function trasformDataForChart(csvData, xColumn, yColumns) {
  return {
    labels: csvData && csvData.data[xColumn],
    datasets:
      csvData &&
      yColumns.map((yColumn, index) => {
        return {
          ...chartStyleConfig,
          label: yColumn,
          borderColor: getColor(index),
          backgroundColor: getColor(index),
          data: csvData && csvData.data[yColumn],
        };
      }),
  };
}

function getYaxisNames(yColumns) {
  return yColumns.map((yColumn) => yColumn.name);
}

export { trasformDataForChart, getYaxisNames };
