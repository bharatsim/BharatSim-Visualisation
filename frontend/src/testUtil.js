// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent, within } from '@testing-library/react';

export function selectDropDownOption(container, dropDownId, optionId) {
  const dropDown = container.getByTestId(dropDownId);
  fireEvent.mouseDown(within(dropDown).getByRole('button'));
  const options = within(
    within(document.getElementById(`menu-${dropDownId}`)).getByRole('listbox'),
  );
  fireEvent.click(options.getByTestId(`${dropDownId}-${optionId}`));
}
