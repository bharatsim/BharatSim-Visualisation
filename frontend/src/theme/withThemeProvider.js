import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
