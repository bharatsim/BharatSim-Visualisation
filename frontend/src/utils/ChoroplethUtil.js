import L from 'leaflet';
import chroma from 'chroma-js';

function highlightOnClick(e) {
  const layer = e.target;

  layer.setStyle({
    weight: 2,
    dashArray: '',
    opacity: 1,
    fillOpacity: 0.7,
  });

  if (!L.Browser.ie && !L.Browser.opera && !L.Browser.edge) {
    layer.bringToFront();
  }
}

function resetHighlight(geoJson) {
  return (e) => geoJson.current.resetStyle(e.target);
}

function zoomToFeature(map) {
  return (e) => map.fitBounds(e.target.getBounds());
}

function geoJSONStyle(idDataMap, mapLayerIdName, maxOfMeasure, scale) {
  const colors = Object.keys(scale)
    .sort((v1, v2) => Number(v1) - Number(v2))
    .map((key) => scale[key]);
  const getColor = chroma.scale(colors);
  return (feature) => {
    const gradientPercentage = idDataMap[feature.properties[mapLayerIdName]] / maxOfMeasure;
    return {
      color: '#1f2021',
      weight: 1,
      fillOpacity: 0.5,
      opacity: 0.7,
      fillColor: getColor(gradientPercentage),
    };
  };
}

export { highlightOnClick, resetHighlight, geoJSONStyle, zoomToFeature };
