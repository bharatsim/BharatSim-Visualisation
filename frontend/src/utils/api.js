import { contentTypes, httpMethods } from '../constants/fetch';
import { url as serviceURL } from './url';
import { fetchData, uploadData } from './fetch';

function headerBuilder({ contentType }) {
  return {
    'content-type': contentType,
  };
}

function formDataBuilder(data) {
  const formData = new FormData();
  data.forEach(({ name, value }) => formData.append(name, value));
  return formData;
}

const api = {
  saveDashboard: async ({ widgets, layout, dashboardId, name, count }) => {
    return uploadData({
      url: serviceURL.DASHBOARD_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets, layout, dashboardId, name, count },
      }),
    });
  },

  addNewDashboard: async ({ name, projectId }) => {
    return uploadData({
      url: serviceURL.INSERT_DASHBOARD,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets: [], layout: [], name, count: 0, projectId },
      }),
      isCustomErrorHandler: true,
    });
  },

  getAllDashBoardByProjectId: async (projectId) =>
    fetchData({
      url: serviceURL.DASHBOARD_URL,
      query: { projectId, columns: ['name', '_id'] },
    }),

  getAllDashBoard: async () => fetchData({ url: serviceURL.DASHBOARD_URL }),

  uploadFileAndSchema: async ({ file, schema, dashboardId }) =>
    uploadData({
      url: serviceURL.DATA_SOURCES,
      data: formDataBuilder([
        { name: 'datafile', value: file },
        { name: 'schema', value: JSON.stringify(schema) },
        { name: 'dashboardId', value: dashboardId },
      ]),
      headers: headerBuilder({ contentType: contentTypes.FILE }),
    }),

  getCsvHeaders: async (dataSourceId) => fetchData({ url: serviceURL.getHeaderUrl(dataSourceId) }),

  getDatasources: async (dashboardId) =>
    fetchData({
      url: serviceURL.DATA_SOURCES,
      query: { dashboardId },
    }),

  getData: async (datasource, columns) =>
    fetchData({ url: serviceURL.getDataUrl(datasource), query: { columns } }),

  saveProject: async ({ id, ...data }) => {
    const requestObject = {
      url: serviceURL.PROJECT_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      isCustomErrorHandler: true,
    };
    if (id) {
      return uploadData({
        ...requestObject,
        data: JSON.stringify({ projectData: { ...data, id } }),
        method: httpMethods.PUT,
      });
    }
    return uploadData({
      ...requestObject,
      data: JSON.stringify({ projectData: { ...data } }),
      method: httpMethods.POST,
    });
  },

  getProjects: async () => {
    return fetchData({
      url: serviceURL.PROJECT_URL,
    });
  },
  getProject: async (id) => {
    return fetchData({
      url: serviceURL.getProjectUrl(id),
    });
  },
};

export { api };
