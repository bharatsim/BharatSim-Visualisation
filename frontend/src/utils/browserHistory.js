// eslint-disable-next-line import/no-mutable-exports
let BrowserHistory;

function initHistory(history) {
  BrowserHistory = history;
}

function navigateToPreviousPage() {
  BrowserHistory.goBack();
}

function navigateToHome() {
  BrowserHistory.push('/');
}

function reloadPage() {
  // eslint-disable-next-line no-restricted-globals
  location.reload();
}

export { initHistory, navigateToPreviousPage, reloadPage, BrowserHistory, navigateToHome };
