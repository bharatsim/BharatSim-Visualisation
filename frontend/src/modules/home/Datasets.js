import React, { useEffect, useState } from 'react';
import { Box, Typography, useTheme } from '@material-ui/core';
import Table from '../../uiComponent/table/Table';
import fileTypes from '../../constants/fileTypes';
import { formatDate } from '../../utils/dateUtils';
import tableStyles from '../../uiComponent/table/tableCSS';
import { convertFileSizeToMB } from '../../utils/helper';
import { api } from '../../utils/api';
import tableIcon from '../../uiComponent/table/tableIcon';
import useModal from '../../hook/useModal';
import DeleteConfirmationModal from '../../uiComponent/DeleteConfirmationModal';

function Datasets() {
  const [dataSources, setDataSources] = useState();
  const theme = useTheme();
  const styles = tableStyles(theme, dataSources);
  const [selectedRow, setSelectedRow] = useState();
  const {
    isOpen: isDeleteModalOpen,
    openModal: openDeleteModal,
    closeModal: closeDeleteModal,
  } = useModal();

  async function fetchDatasets() {
    const { dataSources: fetchedDataSources } = await api.getAllDatasources();
    setDataSources(fetchedDataSources);
  }

  async function deleteDatasource() {
    const { _id: selectedRowId } = selectedRow;
    await api.deleteDatasource(selectedRowId).then(() => {
      setDataSources((prevDatasources) =>
        prevDatasources.filter(({ _id: prevDatasourceID }) => prevDatasourceID !== selectedRowId),
      );
    });
    closeDeleteModal();
  }

  function onDeleteClick(event, rowData) {
    openDeleteModal();
    setSelectedRow(rowData);
  }

  useEffect(() => {
    fetchDatasets();
  }, []);

  if (!dataSources) {
    return null;
  }
  if (dataSources.length === 0) {
    return <Box>No Datasource Found</Box>;
  }
  return (
    <Box>
      <Table
        data={dataSources}
        columns={[
          { title: 'Name', field: 'name' },
          {
            title: 'Size',
            field: 'fileSize',
            type: 'numeric',
            render: (rowData) => convertFileSizeToMB(rowData.fileSize),
          },
          { title: 'Type', field: 'fileType', render: (rowData) => fileTypes[rowData.fileType] },
          {
            title: 'Usage',
            field: 'usage',
            type: 'numeric',
            tooltip: 'Number of Dashboards which has widgets created',
          },
          {
            title: 'Date Added',
            field: 'createdAt',
            type: 'datetime',
            render: (rowData) => formatDate(rowData.createdAt),
          },
        ]}
        title="table"
        options={{
          toolbar: false,
          paging: false,
          sorting: false,
          headerStyle: {
            ...styles.headerStyle,
            color: theme.palette.text.disabled,
            borderTop: 'none',
            borderBottom: `1px solid ${theme.colors.primaryColorScale['500']}3D`,
          },
          actionsColumnIndex: -1,
        }}
        style={{
          ...styles.styles,
        }}
        actions={[
          (rowData) => ({
            icon: tableIcon.Delete,
            tooltip: 'Delete Datasource',
            onClick: onDeleteClick,
            disabled: rowData.usage > 0,
            size: 'small',
            'data-testid': 'delete-datasource',
          }),
        ]}
      />
      <DeleteConfirmationModal
        handleClose={closeDeleteModal}
        title="Delete Datasource"
        open={isDeleteModalOpen}
        deleteAction={{
          onDelete: deleteDatasource,
          name: 'Delete data source',
          dataTestId: 'delete-datasource',
        }}
      >
        <Typography variant="body2">
          {`Are you sure you want to delete datasource ${selectedRow ? selectedRow.name : ''} ?`}
        </Typography>
      </DeleteConfirmationModal>
    </Box>
  );
}

export default Datasets;
