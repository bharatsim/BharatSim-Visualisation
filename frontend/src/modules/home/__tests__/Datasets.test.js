import React from 'react';
import { renderWithRedux as render } from '../../../testUtil';
import withSnackBar from '../../../hoc/snackbar/withSnackBar';
import Datasets from '../Datasets';
import withThemeProvider from '../../../theme/withThemeProvider';
import { api } from '../../../utils/api';

jest.mock('../../../utils/api', () => ({
  api: {
    getAllDatasources: jest.fn().mockResolvedValue({
      dataSources: [
        {
          createdAt: '2020-12-31T09:32:48.123Z',
          dashboardId: 'dashboardId',
          dashboardName: 'DashboardName',
          fileId: 'fileId',
          fileSize: 49440,
          fileType: 'fileType',
          name: 'fileName',
          projectId: 'projectId',
          projectName: 'ProjectName',
          updatedAt: '2020-12-31T09:32:48.123Z',
          _id: 'DatasourceId',
        },
        {
          createdAt: '2020-12-31T09:32:48.123Z',
          fileId: 'fileId',
          fileSize: 49440,
          fileType: 'fileTyp2e',
          name: 'fileName2',
          updatedAt: '2020-12-31T09:32:48.123Z',
          _id: 'DatasourceId2',
        },
        {
          createdAt: '2020-12-31T09:32:48.123Z',
          dashboardId: 'dashboardId',
          dashboardName: 'DashboardName',
          fileId: 'fileId',
          fileSize: 49440,
          fileType: 'fileType',
          name: 'fileName',
          projectId: 'projectId',
          projectName: 'ProjectName',
          updatedAt: '2020-12-31T09:32:48.123Z',
          _id: 'DatasourceId',
        },
      ],
    }),
  },
}));

describe('Datasets', () => {
  const DatasetsWithProvider = withThemeProvider(withSnackBar(Datasets));
  it('should match snapshot Datasets library', async () => {
    const { container, findByText } = render(<DatasetsWithProvider />);
    await findByText('fileName');
    expect(container).toMatchSnapshot();
  });
  it('should show message No Datasource Found if no dataSources are uploaded', async () => {
    api.getAllDatasources.mockResolvedValueOnce({ dataSources: [] });
    const { findByText, getByText } = render(<DatasetsWithProvider />);
    await findByText('No Datasource Found');
    expect(getByText('No Datasource Found')).toBeInTheDocument();
  });
});
