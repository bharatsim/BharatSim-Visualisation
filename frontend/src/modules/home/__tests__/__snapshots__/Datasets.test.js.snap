// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datasets should match snapshot Datasets library 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
      style="position: relative; box-shadow: none; border: 1px solid; border-color: #594ab73d; border-radius: 4px; overflow: hidden;"
    >
      <div
        class="Component-horizontalScrollContainer-11"
        style="overflow-x: auto; position: relative;"
      >
        <div>
          <div
            style="overflow-y: auto;"
          >
            <div>
              <table
                class="MuiTable-root"
                style="table-layout: auto;"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-12 MuiTableCell-alignLeft"
                      scope="col"
                      style="padding: 8px 12px; text-align: left; flex-direction: row; font-family: \\"Rubik\\", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.25; letter-spacing: 0.00938em; color: rgba(52, 58, 64, 0.38); border-bottom: 1px solid #594ab73D; box-sizing: border-box;"
                    >
                      Name
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-12 MuiTableCell-alignRight"
                      scope="col"
                      style="padding: 8px 12px; text-align: left; flex-direction: row; font-family: \\"Rubik\\", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.25; letter-spacing: 0.00938em; color: rgba(52, 58, 64, 0.38); border-bottom: 1px solid #594ab73D; box-sizing: border-box;"
                    >
                      Size
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-12 MuiTableCell-alignLeft"
                      scope="col"
                      style="padding: 8px 12px; text-align: left; flex-direction: row; font-family: \\"Rubik\\", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.25; letter-spacing: 0.00938em; color: rgba(52, 58, 64, 0.38); border-bottom: 1px solid #594ab73D; box-sizing: border-box;"
                    >
                      Type
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-12 MuiTableCell-alignRight"
                      scope="col"
                      style="padding: 8px 12px; text-align: left; flex-direction: row; font-family: \\"Rubik\\", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.25; letter-spacing: 0.00938em; color: rgba(52, 58, 64, 0.38); border-bottom: 1px solid #594ab73D; box-sizing: border-box;"
                    >
                      <span
                        class=""
                        title="Number of Dashboards which has widgets created"
                      >
                        Usage
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-12 MuiTableCell-alignLeft"
                      scope="col"
                      style="padding: 8px 12px; text-align: left; flex-direction: row; font-family: \\"Rubik\\", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.25; letter-spacing: 0.00938em; color: rgba(52, 58, 64, 0.38); border-bottom: 1px solid #594ab73D; box-sizing: border-box;"
                    >
                      Date Added
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-12 MuiTableCell-paddingCheckbox"
                      scope="col"
                      style="padding: 8px 12px; text-align: center; flex-direction: row; font-family: \\"Rubik\\", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.25; letter-spacing: 0.00938em; color: rgba(52, 58, 64, 0.38); border-bottom: 1px solid #594ab73D; width: 48px; box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="true"
                        class="MuiButtonBase-root MuiTableSortLabel-root Mui-disabled"
                        role="button"
                        tabindex="-1"
                      >
                        Actions
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                    index="0"
                    level="0"
                    path="0"
                    style="transition: all ease 300ms; height: 32px; background-color: transparent; border-bottom: 1px solid #594ab73D;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="fileName"
                    >
                      fileName
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="49440"
                    >
                      0.05MB
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="fileType"
                    />
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="1"
                    >
                      1
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="2020-12-31T09:32:48.123Z"
                    >
                      31-Dec-2020 at 9:32 AM
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone"
                      style="width: 48px; padding: 8px 12px; box-sizing: border-box; height: 32px; text-align: left; font-family: \\"Roboto\\", sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1; letter-spacing: 0.01071em;"
                    >
                      <div
                        style="display: flex;"
                      >
                        <span
                          class=""
                          title="Delete Datasource"
                        >
                          <button
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
                            disabled=""
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                    index="1"
                    level="0"
                    path="1"
                    style="transition: all ease 300ms; height: 32px; background-color: rgb(240, 238, 249); border-bottom: 1px solid #594ab73D;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="fileName2"
                    >
                      fileName2
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="49440"
                    >
                      0.05MB
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="fileTyp2e"
                    />
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="0"
                    >
                      0
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="2020-12-31T09:32:48.123Z"
                    >
                      31-Dec-2020 at 9:32 AM
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone"
                      style="width: 48px; padding: 8px 12px; box-sizing: border-box; height: 32px; text-align: left; font-family: \\"Roboto\\", sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1; letter-spacing: 0.01071em;"
                    >
                      <div
                        style="display: flex;"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                          tabindex="0"
                          title="Delete Datasource"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                    index="2"
                    level="0"
                    path="2"
                    style="transition: all ease 300ms; height: 32px; background-color: transparent;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="fileName3"
                    >
                      fileName3
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="49440"
                    >
                      0.05MB
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="fileType"
                    />
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="3"
                    >
                      3
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: 0.875rem; font-family: \\"Roboto\\", sans-serif; font-weight: 400; height: 32px; padding: 8px 12px; text-align: left; line-height: 1; letter-spacing: 0.01071em;"
                      value="2020-12-31T09:32:48.123Z"
                    >
                      31-Dec-2020 at 9:32 AM
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone"
                      style="width: 48px; padding: 8px 12px; box-sizing: border-box; height: 32px; text-align: left; font-family: \\"Roboto\\", sans-serif; font-weight: 400; font-size: 0.875rem; line-height: 1; letter-spacing: 0.01071em;"
                    >
                      <div
                        style="display: flex;"
                      >
                        <span
                          class=""
                          title="Delete Datasource"
                        >
                          <button
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
                            disabled=""
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
